packages/keys/src/key-pair.ts

```diff
@@ -41,3 +42,19 @@ export async function createKeyPairFromBytes(bytes: ReadonlyUint8Array, extracta
 
     return { privateKey, publicKey } as CryptoKeyPair;
 }
+
+export async function createKeyPairFromPrivateKeyBytes(
+    bytes: ReadonlyUint8Array,
+    extractable: boolean = false,
+): Promise<CryptoKeyPair> {
+    const promises = [createPrivateKeyFromBytes(bytes, true)];
+
+    if (!extractable) {
+        promises.push(createPrivateKeyFromBytes(bytes, false));
+    }
+
+    const [extractablePrivateKey, privateKey] = await Promise.all(promises);
+    const publicKey = await getPublicKeyFromPrivateKey(extractablePrivateKey, true);
```

Comment:
This serializes the generation of the public key on the completion of _all_ private key generation. There's a chance this could extend the critical path.

There's a possibility of slightly more runtime efficiency doing this:

```ts
const privateKeyPromise = createPrivateKeyFromBytes(bytes, extractable);
const [publicKey, privateKey] = await Promise.all([
    (extractable ? privateKeyPromise : createPrivateKeyFromBytes(bytes, true /* extractable */)).then(
        async privateKey => await getPublicKeyFromPrivateKey(privateKey, true /* extractable */),
    ),
    privateKeyPromise,
]);
```

If the doubled-extractable private key _happens_ to finish first then you can start the public key generation right away.

Reply:
Love it, thanks!